Raylib-cs/types/Window.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Numerics;
+
+namespace Raylib_cs;
+
+public sealed class Window : IDisposable
+{
+    public event OnInit Init;
+    public event OnUpdate Update;
+    public event OnDraw Draw;
+    public event OnClose Close;
+
+    private int _width;
+    private int _height;
+    private string _title;
+    
+    public Window(int width, int height, string title) {
+        this._width = width;
+        this._height = height;
+        this._title = title;
+    }
+
+    public void Run() {
+        Raylib.InitWindow(_width, _height, _title);
+        Init?.Invoke();
+
+        while (!ShouldClose()) {
+            Update?.Invoke();
+            
+            Raylib.BeginDrawing();
+            Raylib.ClearBackground(Color.SKYBLUE);
+            Draw?.Invoke();
+            Raylib.EndDrawing();
+        }
+        
+        Close?.Invoke();
+        Raylib.CloseWindow();
+    }
```

Comment:
Unfortunately I think this is too drastic of a change to the standard logic of raylib.

I would much rather prefer if I could just do this to get my window opened:
```c#
using Window window = Window.Init(1600, 900, "Hello World!");